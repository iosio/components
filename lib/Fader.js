(function (global, factory) {
    if (typeof define === "function" && define.amd) {
        define(['exports', 'react', 'jss', 'jss-preset-default', 'classnames'], factory);
    } else if (typeof exports !== "undefined") {
        factory(exports, require('react'), require('jss'), require('jss-preset-default'), require('classnames'));
    } else {
        var mod = {
            exports: {}
        };
        factory(mod.exports, global.react, global.jss, global.jssPresetDefault, global.classnames);
        global.Fader = mod.exports;
    }
})(this, function (exports, _react, _jss, _jssPresetDefault, _classnames) {
    'use strict';

    Object.defineProperty(exports, "__esModule", {
        value: true
    });
    exports.Fader = undefined;

    var _react2 = _interopRequireDefault(_react);

    var _jss2 = _interopRequireDefault(_jss);

    var _jssPresetDefault2 = _interopRequireDefault(_jssPresetDefault);

    var _classnames2 = _interopRequireDefault(_classnames);

    function _interopRequireDefault(obj) {
        return obj && obj.__esModule ? obj : {
            default: obj
        };
    }

    function _objectWithoutProperties(obj, keys) {
        var target = {};

        for (var i in obj) {
            if (keys.indexOf(i) >= 0) continue;
            if (!Object.prototype.hasOwnProperty.call(obj, i)) continue;
            target[i] = obj[i];
        }

        return target;
    }

    function _classCallCheck(instance, Constructor) {
        if (!(instance instanceof Constructor)) {
            throw new TypeError("Cannot call a class as a function");
        }
    }

    var _createClass = function () {
        function defineProperties(target, props) {
            for (var i = 0; i < props.length; i++) {
                var descriptor = props[i];
                descriptor.enumerable = descriptor.enumerable || false;
                descriptor.configurable = true;
                if ("value" in descriptor) descriptor.writable = true;
                Object.defineProperty(target, descriptor.key, descriptor);
            }
        }

        return function (Constructor, protoProps, staticProps) {
            if (protoProps) defineProperties(Constructor.prototype, protoProps);
            if (staticProps) defineProperties(Constructor, staticProps);
            return Constructor;
        };
    }();

    function _possibleConstructorReturn(self, call) {
        if (!self) {
            throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
        }

        return call && (typeof call === "object" || typeof call === "function") ? call : self;
    }

    function _inherits(subClass, superClass) {
        if (typeof superClass !== "function" && superClass !== null) {
            throw new TypeError("Super expression must either be null or a function, not " + typeof superClass);
        }

        subClass.prototype = Object.create(superClass && superClass.prototype, {
            constructor: {
                value: subClass,
                enumerable: false,
                writable: true,
                configurable: true
            }
        });
        if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass;
    }

    _jss2.default.setup((0, _jssPresetDefault2.default)());
    var getStyles = function getStyles(styles) {
        var _jss$createStyleSheet = _jss2.default.createStyleSheet(styles).attach(),
            classes = _jss$createStyleSheet.classes;

        return classes;
    };

    var Fader = exports.Fader = function (_React$PureComponent) {
        _inherits(Fader, _React$PureComponent);

        function Fader(props) {
            _classCallCheck(this, Fader);

            var _this = _possibleConstructorReturn(this, (Fader.__proto__ || Object.getPrototypeOf(Fader)).call(this, props));

            var duration = props.duration;

            var value = duration ? duration : 175;
            var styles = {
                root: {
                    opacity: 0
                },
                trans: {
                    transition: 'opacity ' + value + 'ms ease-in-out'
                },
                show: {
                    opacity: 1
                }
            };
            _this.classes = getStyles(styles);
            return _this;
        }

        _createClass(Fader, [{
            key: 'render',
            value: function render() {
                var classes = this.classes;

                var _props = this.props,
                    className = _props.className,
                    style = _props.style,
                    show = _props.show,
                    children = _props.children,
                    ref = _props.ref,
                    rest = _objectWithoutProperties(_props, ['className', 'style', 'show', 'children', 'ref']);

                return _react2.default.createElement(
                    'div',
                    Object.assign({
                        ref: ref,
                        className: (0, _classnames2.default)(classes.root, classes.trans, show ? classes.show : null, className),
                        style: Object.assign({}, style)
                    }, rest),
                    children
                );
            }
        }]);

        return Fader;
    }(_react2.default.PureComponent);

    ;
});