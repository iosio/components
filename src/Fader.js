import React from 'react';
export class Fader extends React.PureComponent {
    render() {
        const { className, style, show, children, duration, ref, ...rest } = this.props;
        const value = duration ? duration : 175;
        return (
            <div
                id="Fader"
                ref={ref}
                className={className}
                style={{
                    ...style,
                    opacity: show ? 1: 0,
                    transition: `opacity ${value}ms ease-in-out`
                }}
                {...rest}>

                {children}
            </div>
        );
    }
}









