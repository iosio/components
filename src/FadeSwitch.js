import React from 'react';

import {Fader} from "./Fader";

export class FadeSwitch extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            show: false,
            view: 0,
            initial_load: true,
        };
        this.is_mounted_ = false;
        this.timeout = false;
        this.duration = props.duration || props.duration === 0 ? props.duration : 175;
    }

    componentDidMount() {
        this.is_mounted_ = true;
        this.showView(this.props.view);
    }

    showView = (view_num) => {

        if (this.is_mounted_) {

            if (!this.state.initial_load) {

                this.setState({show: false});

                this.timeout = setTimeout(() => {
                    if (this.is_mounted_) {
                        this.setState({view: view_num, show: true,});
                    }
                }, this.duration);

            } else {
                this.setState({view: view_num, show: true, initial_load: false});
            }
        }
    };

    componentDidUpdate(prevProps) {
        if (prevProps.view !== this.props.view) {
            this.showView(this.props.view);
        }
    }

    componentWillUnmount() {
        this.is_mounted_ = false;
        clearTimeout(this.timeout);
    }

    render() {
        const {
            View0,
            View1,
            className,
            fader_props,
            style
        } = this.props;

        const {
            show,
            view,
        } = this.state;

        let fp_style = {};
        if (fader_props && fader_props.style) {
            fp_style = fader_props.style;
        }
        const combined_style = {...style, ...fp_style};

        return (
            <Fader
                duration={this.duration}
                className={className}
                style={{
                    height: '100%',
                    width: '100%',
                    position: 'relative',
                    ...combined_style,
                }}
                show={show}>

                {view === 0 ? View0 : View1}
            </Fader>

        );
    };
}
