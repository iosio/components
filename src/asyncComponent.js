import React from 'react';
import {FadeSwitch} from "./FadeSwitch";

export const asyncComponent = (getComponent, Loader, duration, fadeSwitch_props, fader_props) => {
    return class AsyncComponent extends React.Component {
        constructor() {
            super();
            if (!this.state.Component) {
                getComponent().then(Component => {
                    Component = Component && Component.__esModule ? Component.default : Component;
                    AsyncComponent.Component = Component;
                    this.setState({Component});
                });
            }
        }

        static Component = null;
        state = {Component: AsyncComponent.Component};

        render() {
            const {Component} = this.state;

            return (
                <React.Fragment>
                    {Loader && duration > 0 ? <FadeSwitch
                            {...fadeSwitch_props}
                            fader_props={fader_props}
                            duration={duration}
                            {...this.props}
                            style={{height: '100%', width: '100%'}}
                            view={!!Component ? 1 : 0}
                            View0={Loader ? <Loader/> : null}
                            View1={Component ? <Component {...this.props}/> : null}
                        />
                        :

                        (Component ? <Component {...this.props}/> : null)
                    }
                </React.Fragment>
            );
        }
    }
};


