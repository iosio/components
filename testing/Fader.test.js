import React from 'react';
import render from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {Fader} from "../src/Fader";

Enzyme.configure({adapter: new Adapter()});

import {mount, shallow} from 'enzyme';



describe('Fader Component', ()=>{

    it('should match snapshot when show = true', () => {

        const tree = render.create(
            <Fader show={true}>
                <div>derp</div>
            </Fader>
        );

        expect(tree).toMatchSnapshot();

    });


    it('should match snapshot when show = false', () => {

        const tree = render.create(
            <Fader show={false}>
                <div>derp</div>
            </Fader>
        );

        expect(tree).toMatchSnapshot();

    });


    it('should match snapshot when duration is configured', () => {

        const tree = render.create(
            <Fader duration={500}>
                <div>derp</div>
            </Fader>
        );

        expect(tree).toMatchSnapshot();

    });


});