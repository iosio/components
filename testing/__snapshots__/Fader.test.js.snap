// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fader Component should match snapshot when duration is configured 1`] = `
<div
  id="Fader"
  style={
    Object {
      "opacity": 0,
      "transition": "opacity 500ms ease-in-out",
    }
  }
>
  <div>
    derp
  </div>
</div>
`;

exports[`Fader Component should match snapshot when show = false 1`] = `
<div
  id="Fader"
  style={
    Object {
      "opacity": 0,
      "transition": "opacity 175ms ease-in-out",
    }
  }
>
  <div>
    derp
  </div>
</div>
`;

exports[`Fader Component should match snapshot when show = true 1`] = `
<div
  id="Fader"
  style={
    Object {
      "opacity": 1,
      "transition": "opacity 175ms ease-in-out",
    }
  }
>
  <div>
    derp
  </div>
</div>
`;
